dotnet/src/Experimental/Process.Runtime.Dapr/Actors/ProcessActor.cs

```diff
@@ -94,11 +94,11 @@ public Task StartAsync(bool keepAlive)
     /// </summary>
     /// <param name="processEvent">Required. The <see cref="KernelProcessEvent"/> to start the process with.</param>
     /// <returns>A <see cref="Task"/></returns>
-    public async Task RunOnceAsync(KernelProcessEvent processEvent)
+    public async Task RunOnceAsync(string processEvent)
```

Comment:
wondering if there should be an overload, since it will be different from the interface in LocalProcess, but also I guess in DAPR this is abstracted from the user so it's ok (?)

Reply:
Yes, in my _mental map_ also the implementation details between local-runtime and dapr-runtime seem to diverge quite a lot with the only public surface area being the extension methods on the `*Factory` classes (`DaprKernelProcessFactory` / `LocalKernelProcessFactory`) and also the `*Context` classes that share a common base: `KernelProcessContext`.

Since (for Dapr) calling the remote actor crosses the remoting boundary, we must serialize the event.

Ultimately, `ProcessActor` is internal.

Reply:
side note, maybe in the official documentation we can have a nice table showing the equivalent components in different runtimes